agent/handlers/task_server_setup.go

```diff
@@ -110,9 +112,11 @@ func v3HandlersSetup(muxRouter *mux.Router,
 	state dockerstate.TaskEngineState,
 	statsEngine stats.Engine,
 	cluster string,
-	availabilityZone string) {
+	availabilityZone string,
+	containerInstanceArn string) {
 	muxRouter.HandleFunc(v3.ContainerMetadataPath, v3.ContainerMetadataHandler(state))
-	muxRouter.HandleFunc(v3.TaskMetadataPath, v3.TaskMetadataHandler(state, cluster, availabilityZone))
+	muxRouter.HandleFunc(v3.TaskMetadataPath, v3.TaskMetadataHandler(state, cluster, availabilityZone, containerInstanceArn, false))
+	muxRouter.HandleFunc(v3.TaskWithTagsMetadataPath, v3.TaskMetadataHandler(state, cluster, availabilityZone, containerInstanceArn, true))
```

Comment:
why we only expose tags through v3 metadata endpoint?

Reply:
v2 is a subset of v3 but only support awsvpc mode. I've updated v2 endpoint to include a /v2/metadataWithTags endpoint